src/Raven.Client/Util/TcpUtils.cs

```diff
@@ -144,5 +144,34 @@ private static TcpClient NewTcpClient(TimeSpan? timeout, bool useIPv6)
             Debug.Assert(tcpClient.Client != null);
             return tcpClient;
         }
+
+        internal static async Task<(TcpClient Client, string Url)> ConnectAsyncWithPriority(TcpConnectionInfo info, TimeSpan? tcpConnectionTimeout)
+        {
+            string selectedUrl = null;
+            TcpClient tcpClient = null;
+            
+            foreach (var url in info.TcpServerUrls)
+            {
+                try
+                {
+                    tcpClient = await ConnectAsync(url, tcpConnectionTimeout).ConfigureAwait(false);
+                    selectedUrl = url;
+                    break;
```

Comment:
Just return here, no?

Reply:
for the case where the original server url was not tested we test it below this might mean we will try it twice if it appears in the url list, we could check if it contains it and return

Reply:
Unless I miss something, this function returns the first working url.
I don't see any double checks here

Reply:
you're correct